pkg/controller/tas/node_failure_controller.go

```diff
@@ -187,30 +195,64 @@ func (r *nodeFailureReconciler) patchWorkloadsForUnavailableNode(ctx context.Con
 			}
 			continue
 		}
-
+		annotations := getAnnotations(&wl)
+		if failedNode, ok := annotations[kueuealpha.NodeToReplaceAnnotation]; ok && failedNode != nodeName && !workload.IsEvicted(&wl) {
+			r.log.V(3).Info("Evicting workload due to multiple node failures", "workload", wlKey)
+			evictionMsg := fmt.Sprintf(NodeMultipleFailuresEvictionMessageFormat, failedNode, nodeName)
+			if evictionErr := r.startEviction(ctx, &wl, evictionMsg); evictionErr != nil {
+				r.log.Error(evictionErr, "Failed to complete eviction process", "workload", wlKey)
+				workloadProcessingErrors = append(workloadProcessingErrors, evictionErr)
+			}
+			if err := r.client.Get(ctx, wlKey, &wl); err != nil {
+				r.log.V(2).Error(err, "Failed to re-fetch workload after eviction", "workload", wlKey)
+				workloadProcessingErrors = append(workloadProcessingErrors, err)
+			} else {
+				annotations = getAnnotations(&wl)
+			}
+		}
 		err := clientutil.Patch(ctx, r.client, &wl, true, func() (bool, error) {
-			currentAnnotations := wl.GetAnnotations()
-			if currentAnnotations == nil {
-				currentAnnotations = make(map[string]string)
+			failedNode, ok := annotations[kueuealpha.NodeToReplaceAnnotation]
+			if !ok {
+				r.log.V(4).Info(fmt.Sprintf("Adding node to %s annotation", kueuealpha.NodeToReplaceAnnotation), "workload", wlKey, "nodeName", failedNode)
+				annotations[kueuealpha.NodeToReplaceAnnotation] = nodeName
+				wl.SetAnnotations(annotations)
+				return true, nil
 			}
-			existingFailedNode, ok := currentAnnotations[kueuealpha.NodeToReplaceAnnotation]
-			if ok && existingFailedNode == nodeName {
-				return false, nil
+			if workload.IsEvicted(&wl) {
+				r.log.V(4).Info(fmt.Sprintf("Removing node from %s annotation", kueuealpha.NodeToReplaceAnnotation), "workload", wlKey, "nodeName", failedNode)
+				delete(annotations, kueuealpha.NodeToReplaceAnnotation)
+				wl.SetAnnotations(annotations)
+				return true, nil
 			}
-			currentAnnotations[kueuealpha.NodeToReplaceAnnotation] = nodeName
-			wl.SetAnnotations(currentAnnotations)
-			r.log.V(4).Info("Adding unavailable node to workload annotation", "workload", wlKey, "nodeName", nodeName)
-			return true, nil
+			return false, nil
 		})
 		if err != nil {
-			r.log.V(2).Error(err, "Failed to patch workload with annotation", "workload", wlKey)
+			r.log.V(2).Error(err, "Failed to patch workload annotation", "workload", wlKey)
 			workloadProcessingErrors = append(workloadProcessingErrors, err)
 		} else {
-			r.log.V(3).Info("Successfully patched workload with annotation", "workload", wlKey, "unavailableNodesAnnotation", wl.GetAnnotations()[kueuealpha.NodeToReplaceAnnotation])
+			r.log.V(3).Info("Successfully patched workload annotation", "workload", wlKey, "nodesToReplaceAnnotation", wl.GetAnnotations()[kueuealpha.NodeToReplaceAnnotation])
 		}
```

Comment:
Similarly here, maybe we could `continue` in case of error and get rid of `else` statement